### PR TITLE
feat: reset-totp inside control-center

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
image

## Checklist

- [X] I ran `npm run re:build`
- [X] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    try {
+      open LogicUtils
+      setScreenState(_ => PageLoaderWrapper.Loading)
+      let url = getURL(
+        ~entityName=USERS,
+        ~userType=#CHECK_TWO_FACTOR_AUTH_STATUS,
+        ~methodType=Get,
+        (),
+      )
+      let res = await fetchDetails(url)
```
Comment (JeevaRamu0104) [inline src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res:161]: define type for this response

File: src/entryPoints/HyperSwitchApp.res [hunk 2]
```diff
@@ -460,7 +460,13 @@ let make = () => {
                             permission=userPermissionJson.merchantDetailsManage>
                             <HSwitchSettings />
                           </AccessControl>
-                        | list{"account-settings", "profile"} => <HSwitchProfileSettings />
+                        | list{"account-settings", "profile", ...remainingPath} =>
+                          <EntityScaffold
+                            entityName="ConfigurePMTs"
+                            remainingPath
+                            renderList={() => <HSwitchProfileSettings />}
+                            renderShow={value => <ModifyTwoFaSettings />}
```
Comment (JeevaRamu0104) [inline src/entryPoints/HyperSwitchApp.res:468]: wrap the <ModifyTwoFaSettings /> with the featureflag totp

File: src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        generateNewSecret()->ignore
+      } else {
+        setShowVerifyModal(_ => true)
+      }
+      None
+    })
+
+    <div>
+      <Modal
+        modalHeading="Verify OTP"
```
Comment (JeevaRamu0104) [inline src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res:87]: It cannot always be Verify Totp.we need to allow user to enter recovery code also

File: src/entryPoints/HyperSwitchApp.res (hunk 2 above)
Comment (Riddhiagrawal001) [reply src/entryPoints/HyperSwitchApp.res]: Wrapped  <ModifyTwoFaSettings /> with featureflag as the whole component cannot be wrapped

File: src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res (hunk 3 above)
Comment (Riddhiagrawal001) [reply src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res:162]: Added the feature to add access code as well here

File: src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+        let res = await fetchDetails(url)
+        setTotpUrl(_ =>
+          res->getDictFromJsonObject->getDictfromDict("secret")->getString("totp_url", "")
+        )
+        setShowRegenSecret(_ => false)
+      } catch {
+      | Exn.Error(e) => {
+          let err = Exn.message(e)->Option.getOr("Verification Failed")
+          let errorCode = err->safeParse->getDictFromJsonObject->getString("code", "")
+          if errorCode === "UR_40" {
```
Comment (JeevaRamu0104) [inline src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res:95]: can you please map this error code to type

File: src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+        } else {
+          showToast(~message="OTP field cannot be empty!", ~toastType=ToastError, ())
+        }
+        setButtonState(_ => Button.Normal)
+      } catch {
+      | Exn.Error(e) => {
+          let err = Exn.message(e)->Option.getOr("Verification Failed")
+          let errorMessage = err->safeParse->getDictFromJsonObject->getString("message", "")
+          let errorCode = err->safeParse->getDictFromJsonObject->getString("code", "")
+          if errorCode === "UR_42" {
```
Comment (JeevaRamu0104) [inline src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res:129]: same here

**Pull request merged**